### PR TITLE
Link equivalent fr page

### PR DESCRIPTION
**Pull request by samjewell**
This combines the country-picker popup work with @Ken-Lo's work to add country specific metadata to the page headers. This means that /pricing links straight to /fr/tarifs, and back again, etc.

screen shot 2015-02-02 at 20 07 10

screen shot 2015-02-02 at 20 09 56

### CODE REVIEW

File: app/includes/_site_footer.html [hunk 1]
```diff
@@ -73,18 +74,42 @@
...
-      </p>
-      <p>
-        You can now accept payments from across Europe with both <a href="/features" id="track-footer-bottom-features" class="u-link-clean u-link-invert">GoCardless</a> and <a href="/pro" id="track-footer-bottom-pro" class="u-link-clean u-link-invert">GoCardless Pro</a>.
-      </p>
-      <div class="u-color-invert u-text-upcase u-text-xxs u-text-heading u-text-semi u-margin-Txs">
-        <a href="/europe" id="track-footer-bottom-europe" class="u-link-clean u-link-invert">Find out more</a>
-      </div>
+      <span class="u-text-light u-text-no-smoothing">Currently viewing</span>
+      <img src="/images/icons/uk-flag-icon@2x.png" class="flag-icon u-margin-Hs" alt="">
+      <span class="u-relative">
```
Comment (samjewell) [inline app/includes/_site_footer.html]: @jamesshedden This u-relative class is required for positioning of the popover. Do you think it's fine to keep it as this utility class, or would you name it as `popover-wrapper`, or `nav` (and `popover` become `nav-popover`) or something like that?

File: app/includes/fr/_site_footer.html [hunk 2]
```diff
@@ -37,4 +38,41 @@
       </div>
     </div>
   </div>
+  <div class="page-footer__end u-text-center u-text-heading u-text-xxs u-color-invert u-padding-Vl u-margin-Tl" id="track-footer-end-links">
+    <div class="u-padding-Vl">
+      <span class="u-text-light u-text-no-smoothing">Vous consultez actuellement</span>
+      <img src="/images/icons/fr-flag-icon@2x.png" class="flag-icon u-margin-Hs" alt="">
+      <span class="u-relative">
```
Comment (samjewell) [inline app/includes/fr/_site_footer.html]: same here @jamesshedden

**Comment by samjewell** [comment]
@Ken-Lo do you want to give this a final check-over?
@jamesshedden just wondered if you're happy with where this has got to? Thanks so much both :cake: :rose: :camel: 
The Popover now seems to take 2 clicks before it opens for the first time, since incorporating Kin-Hing's changes. Any ideas why this might be happening @jamesshedden or @harrison ?
Introduced `id="track-footer-links"` onto the whole footer div with this PR - will remove all of the individual IDs at a later date when they've switched the analytics over.

**Comment by Ken-Lo** [comment]
Hi guys,
I've just added some notes to https://github.com/gocardless/splash-pages/pull/405 so you're able to see some background on the hreflang tag.
@samjewell Will take a look at your code later this eve!

File: app/includes/fr/_site_footer.html [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
-  <div class="grid site-container u-padding-Vl u-text-center page-footer__start">
+<div class="page-footer u-color-invert u-padding-Tl">
+  <div class="grid site-container u-padding-Vl u-text-center page-footer__start"
+  id="track-footer-links">
     <div class="grid__cell u-size-1of2">
-      <ul class="grid u-text-upcase u-text-heading u-text-semi">
+      <ul class="grid u-text-heading u-text-semi">
         <div class="grid__cell u-text-xxs">
           <li class="u-pull-start u-margin-Rl">
             <a href="/fr/a-propos?l=2" class="u-link-clean u-link-invert">A propos</a>
```
Comment (Ken-Lo) [inline app/includes/fr/_site_footer.html]: We can remove ?l=2 now - we aren't using it.

File: app/pages/fr/securite/index.html [hunk 4]
```diff
@@ -1,3 +1,11 @@
+---
+language: fr
+other_languages:
+  en: /security
+---
+
+{{other_languages.fr}}
```
Comment (Ken-Lo) [inline app/pages/fr/securite/index.html]: Whoops - I accidentally left this in the pull request - any chance you could take this line out?

**Comment by Ken-Lo** [comment]
Hi Sam,
This looks really good to me - only two small things (sorry, both are my bad):
1. I left some debugging code on the French security pages.
2. For completeness (this is kind of related to my original PR rather than yours but seeing as we're including the two PRs in one), I noticed the HTML code still outputs `<html lang="en">` regardless of whether the splash page is in English or French (lines 2 and 3 of `apps/templates/plain.html`). Can we change lines 2 and 3 from `lang="en"` to `lang="{{ language if language else 'en' }">`? It should then output `<html lang="fr">` on all of the French splash pages.
Thanks again!
Kin-Hing

File: app/templates/plain.html [hunk 5]
```diff
@@ -11,6 +11,18 @@
...
     <meta name="google-site-verification" content="Y80kah87ghJhwiDqw-5ap234p9wCcGt6kMRxvnamtHU">
     <link href="https://plus.google.com/+Gocardless" rel="publisher">
+    {%- macro hreflang_tag(language_code, path) -%}
+      <link rel="alternate" href="https://gocardless.com{{path}}" hreflang="{{language_code}}" />
+      {% if language_code == "en" -%}
+        <link rel="alternate" href="https://gocardless.com{{path}}" hreflang="x-default" />
+      {%- endif %}
+    {%- endmacro -%}
+    {% if language %}
+      {{ hreflang_tag(language, path) }}
```
Comment (harrison) [inline app/templates/plain.html]: Duplicate?

File: scripts/nunjucks.js [hunk 6]
```diff
@@ -31,6 +31,11 @@ function stream(filepath, outfile, compile) {
   }, function(file, enc, cb) {
     var _this = this;
     var parsed = frontMatter(file.toString());
+	
+	// Strip output folder and trailing /index.html to get the path on website
```
Comment (harrison) [inline scripts/nunjucks.js]: Would pass the outfile and outroot to the stream function instead of reading it from argv again

Comment (harrison) [inline scripts/nunjucks.js]: Also indentation and single quotes

File: app/includes/fr/_site_footer.html (hunk 3 above)
Comment (samjewell) [inline app/includes/fr/_site_footer.html]: This already got merged in PR https://github.com/gocardless/splash-pages/pull/401/files so should be fine

File: scripts/nunjucks.js (hunk 6 above)
Comment (samjewell) [inline scripts/nunjucks.js]: @harrison sorry, not sure exactly what you mean about outroot and outfile... :-( I'll grab you

**Comment by Ken-Lo** [comment]
@harrison @samjewell Thanks for tidying up guys!

With regards to `app/templates/plain.html`, the original code wasn't actually a duplicate (see my pull request https://github.com/gocardless/splash-pages/pull/405 for more information).

The hreflang_tag macro outputs the `<link rel="alternative" hreflang>` tag given two arguments: a language code an the path. If the language_code is equal to en, it also outputs a tag with hreflang="x-default" (see [example 3 on the Moz blog post here](http://moz.com/blog/hreflang-behaviour-insights)).

The bit below it actually calls the macro: first with the language of the current page and then looping through the `other_languages` variable.

File: app/templates/plain.html [hunk 7]
```diff
@@ -11,6 +11,12 @@
     <meta name="og:image:secure_url" content="https://gocardless.com/images/logos/gocardless-square.png">
     <meta name="google-site-verification" content="Y80kah87ghJhwiDqw-5ap234p9wCcGt6kMRxvnamtHU">
     <link href="https://plus.google.com/+Gocardless" rel="publisher">
+    {%- macro hreflang_tag(language_code, path) -%}
```
Comment (harrison) [inline app/templates/plain.html]: how does this work now?

File: scripts/nunjucks.js [hunk 8]
```diff
@@ -102,12 +107,13 @@ findFiles((argv.i || argv.input), function(filepath) {
   var outfile = path.join(cwd, filepath).replace(inputCwd, '');
   var output = argv.o || argv.output;
   outfile = path.join(output, outfile);
+
   mkdirp.sync(path.dirname(outfile));
 
   console.log('Compiling %s (%s)', filepath, outfile);
 
   stream(filepath, outfile, function(file, fileMetadata) {
```
Comment (harrison) [inline scripts/nunjucks.js]: naming is a bit confusing here, would probably change `filepath > inFile`, `outfile > outFile`, `output > outRoot`

then pass `outRoot` `stream(inFile, outFile, outRoot, function...)`

File: app/templates/plain.html (hunk 7 above)
Comment (samjewell) [inline app/templates/plain.html]: It doesn't. If you take out lines 20 - 25 then the macro never gets called, and these 3 lines (below) disappear from the `<head>` of pages such as /pro
```
<link rel="alternate" href="https://gocardless.com/pro" hreflang="en">
<link rel="alternate" href="https://gocardless.com/pro" hreflang="x-default">
<link rel="alternate" href="https://gocardless.com/fr/pro" hreflang="fr">
```

File: scripts/nunjucks.js (hunk 8 above)
Comment (samjewell) [inline scripts/nunjucks.js]: You'd do that renaming through the whole nunjucks file right?

Comment (harrison) [inline scripts/nunjucks.js]: :+1:

**Comment by samjewell** [comment]
@harrison passing tests and freed from merge conflicts. Do you want to have a final check over it? :banana: :apple:

**Comment by harrison** [comment]
:+1:

**Comment by samjewell** [comment]
@harrison @Ken-Lo WOOOOOP!!! :boom: :boom: :boom: :boom: :boom:

**Pull request merged**